### PR TITLE
Fixed ragdoll panel being available in Animation Editor when PhysX gem is enabled.

### LINKED ISSUE #2540: Ragdoll editor check for PhysX gem fails
**Describe the bug**
The ragdoll editor in the animation editor checks to see if the PhysX gem is enabled, but it does so by checking for a particular TypeId which no longer exists, so it always thinks PhysX is disabled.

**To Reproduce**
1. Open the animation editor
2. Select physics in the dropdown
3. Editor reports that PhysX gem is disabled, even if it is enabled.

**Expected behavior**
Ragdoll editing is available provided PhysX gem is enabled (even that behaviour is not ideal because it makes a soft dependency on PhysX)

### PR DESCRIPTION
**Pull request by moraaar**
Following the same approach as cloth plugin, which is to ask if the system component of the gem is available (in this case PhysX::SystemComponent).

Fixes #2540

Signed-off-by: moraaar <moraaar@amazon.com>

### CODE REVIEW

**Review by amzn-sean — APPROVED** [review]
LGTM

**Review by hultonha — APPROVED** [review]
LGTM

**Review by SergeyAMZN — APPROVED** [review]

File: Gems/EMotionFX/Code/Tests/D6JointLimitConfiguration.h [hunk 1]
```diff
@@ -14,14 +14,13 @@ namespace EMotionFX
...
-    class D6JointLimitConfiguration
+    struct D6JointLimitConfiguration
         : public AzPhysics::JointConfiguration
     {
     public:
         AZ_CLASS_ALLOCATOR(D6JointLimitConfiguration, AZ::SystemAllocator, 0);
         // This uses the same uuid as the production D6JointLimitConfiguration.
-        // The Ragdoll UI uses this UUID to see if physx is available.
-        AZ_RTTI(D6JointLimitConfiguration, "{90C5C23D-16C0-4F23-AD50-A190E402388E}", AzPhysics::JointConfiguration);
+        AZ_RTTI(D6JointLimitConfiguration, "{88E067B4-21E8-4FFA-9142-6C52605B704C}", AzPhysics::JointConfiguration);
```
Comment (amzn-sean) [inline Gems/EMotionFX/Code/Tests/D6JointLimitConfiguration.h:23]: why the change in the GUID?

Comment (moraaar) [reply Gems/EMotionFX/Code/Tests/D6JointLimitConfiguration.h:23]: The GUID is suppose to match the one from PhysX gem. They used to, but the GUID must have changed in PhysX Gem and was never updated here. Now they match again.

Comment (amzn-sean) [reply Gems/EMotionFX/Code/Tests/D6JointLimitConfiguration.h:23]: that seems flacky.
Is it possible to remove this class? (maybe in another PR)

**Pull request merged**